Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -137,15 +141,41 @@ public interface IonSerializerFactory<T, TContext> where TContext : IonSerializa
     public class IonSerializer
     {
         private readonly IonSerializationOptions options;
+        private Dictionary<Type, dynamic> primitiveSerializers { get; init; }
 
         public IonSerializer() : this(new IonSerializationOptions())
         {
-
         }
 
         public IonSerializer(IonSerializationOptions options)
         {
             this.options = options;
+            this.primitiveSerializers = new Dictionary<Type, dynamic>()

```

Comment:
If these are stateless, then we should just have a read-only static instance of the primitives rather than instantiating per-class.